src/main/java/com/wintertodt/scouter/ui/condensed/WintertodtScouterCondensedPluginPanel.java

```diff
@@ -299,12 +302,43 @@ public void populate(List<WintertodtBossData> globalBossData)
 		List<WintertodtBossData> sortedBossData = new ArrayList<>(globalBossData);
 		sortedBossData.sort(comparator);
 
+		// Get WorldResult once and null-check
+		WorldResult worldResult = plugin.getWorldService().getWorlds();
+		// getWorlds() may return null if the world list has not yet been loaded by RuneLite.
+		// This can occur on slow network connections or immediately after client startup.
+		// In this case, we show a loading indicator until the world list becomes available.
+		if (worldResult == null) {
+			listContainer.removeAll();
+			listContainer.setLayout(new BorderLayout());
+
+			if (loadingPanelCache == null) {
+				loadingPanelCache = new JPanel(new GridBagLayout());
```

Comment:
Let's put this initialization in the constructor with the other component initialization.